src/components/Tool/FixPhoneNumbers/helper.ts

```diff
@@ -1,27 +1,33 @@
 import { PersonUpdateInput } from 'src/graphql/types.generated';
-import { FormValuesPerson } from './FixPhoneNumbers';
+import { PhoneNumberData } from './Contact';
 
 export const determineBulkDataToSend = (
-  values: FormValuesPerson[],
+  dataState: {
+    [key: string]: PhoneNumberData;
+  },
   defaultSource: string,
+  appName: string,
 ): PersonUpdateInput[] => {
   const dataToSend = [] as PersonUpdateInput[];
 
-  values.forEach((value) => {
-    const primaryNumber = value.phoneNumbers.nodes.find(
-      (number) => number.source === defaultSource,
+  Object.entries(dataState).forEach(([id, data]) => {
+    const primaryNumber = data.phoneNumbers.find(
+      (number) =>
+        number.source === defaultSource ||
+        (defaultSource === appName && number.source === appName),
```

Comment:
I think this second condition is redundant because if `defaultSource` and `number.source` both equal `appName`, they are equal to each other.

```suggestion
        number.source === defaultSource,
```